### PR TITLE
Connected components dev

### PR DESCRIPTION
**Pull request by RobinL**

### CODE REVIEW

**Review by RobinL — COMMENTED** [review]
A bunch of comments.  Some of them may just reflect my imperfect understanding of the code

Note that I've since split out the `SplinkDataFrame` into its own file, hence the merge conflicts

File: splink/connected_components.py [hunk 1]
```diff
@@ -0,0 +1,281 @@
+class ConnectedComponents:
+    def __init__(self, linker):
+        self.linker = linker
+
+        if self.linker.settings_dict["link_type"] != "dedupe_only":
+            return ["Not currently implemented"]
+
+        # find our final model
+        predict_dfs = list(
```
[the remark this answers is not in the document]
Comment (Th368MoJ) [reply splink/connected_components.py:9]: This is just a temporary addition for now.

File: splink/connected_components.py [hunk 2]
```diff
@@ -0,0 +1,281 @@
+class ConnectedComponents:
+    def __init__(self, linker):
+        self.linker = linker
+
+        if self.linker.settings_dict["link_type"] != "dedupe_only":
```
Comment (Th368MoJ) [inline splink/connected_components.py:5]: Settings object has the ability to pull the unique id columns directly.

**Comment by Th368MoJ** [comment]
Superseded by:
* [Pulled request #444](https://github.com/moj-analytical-services/splink/pull/444)
* [Pull request #440](https://github.com/moj-analytical-services/splink/pull/440)
* [Pull request #439](https://github.com/moj-analytical-services/splink/pull/439)

**Pull request closed without merging**